R/tmb-methods.R

```diff
@@ -40,43 +40,146 @@ fitted.mmrm_tmb <- function(object, ...) {
   fitted_col[, 1L, drop = TRUE]
 }
 
+#' @describeIn mmrm_tmb_methods predict conditional means for new data;
+#'  optionally with standard errors and confidence or prediction intervals.
+#'  Returns a vector of predictions if `se.fit == FALSE` and
+#'  `interval == "none"`; otherwise it returns a data.frame with multiple
+#'  columns and one row per input data row.
+#' @importFrom stats predict
+#' @exportS3Method
+#' @examples
+#' stop("implement example")
+predict.mmrm_tmb <- function(
+    object, newdata, se.fit = FALSE, # nolint
+    interval = c("none", "confidence", "prediction"), level = 0.95,
+    na.action = na.pass, n_sim = 1000L, ...) { # nolint
+  interval <- match.arg(interval)
+  # make sure new data has the same levels as original data
```

Comment:
This comment is not easy to understand now in the code context, is it even still needed? maybe not?

Reply:
it is to annotate line 58, where we add another "model.frame" prior to `h_mmrm_tmb_data` (as `h_mmrm_tmb_data` does not know the original data).

so basically it is saying that the seemingly unnecessary `model.frame` call is needed